packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx

```diff
@@ -1,99 +1,206 @@
+                : createDashboardFilterRuleFromField(
+                      field,
+                      tilesSavedQueryFilters,
+                  ),
         );
 
-    const filterType = field
-        ? getFilterTypeFromField(field)
-        : FilterType.STRING;
-    const filterConfig = useMemo(
-        () => FilterTypeConfig[filterType],
-        [filterType],
+    const handleChangeFilterRule = useCallback(
+        (newFilterRule: DashboardFilterRule) => {
+            setInternalFilterRule(newFilterRule);
+        },
+        [],
+    );
+
+    const handleChangeFilterOperator = useCallback(
+        (operator: FilterRule['operator']) => {
+            setInternalFilterRule((prevState) =>
+                getFilterRuleWithDefaultValue(field, {
+                    ...prevState,
+                    operator: operator,
+                }),
+            );
+        },
+        [field],
+    );
+
+    const handleChangeTileConfiguration = useCallback(
+        (
+            action: FilterActions,
+            tileUuid: string,
+            filterUuid?: FilterableField,
+        ) => {
+            const savedQuery = tilesSavedQueryFilters[tileUuid];
+
+            setInternalFilterRule((prevState) =>
+                produce(prevState, (draftState) => {
+                    draftState.tileTargetOverride =
+                        draftState.tileTargetOverride?.filter((tileConfig) => {
+                            return tileConfig.tileUuid !== tileUuid;
+                        }) || [];
+
+                    const filters = savedQuery;
+
+                    if (action === FilterActions.ADD) {
+                        const filterableField =
+                            filterUuid ??
+                            filters.find(matchFieldExact(field)) ??
+                            filters.find(matchFieldByTypeAndName(field)) ??
+                            filters.find(matchFieldByType(field));
+
+                        if (!filterableField) return draftState;
+
+                        draftState.tileTargetOverride.push({
+                            tileUuid,
+                            fieldId: fieldId(filterableField),
+                            tableName: filterableField.table,
+                        });
+                    }
+                }),
+            );
+        },
+        [field, tilesSavedQueryFilters],
+    );
+
+    const filterSettings = (
+        <FilterSettings
+            field={field}
+            filterRule={internalFilterRule}
+            onChangeFilterOperator={handleChangeFilterOperator}
+            onChangeFilterRule={handleChangeFilterRule}
+            popoverProps={popoverProps}
+        />
     );
 
     return (
         <ConfigureFilterWrapper>
-            {onBack && (
-                <BackButton minimal onClick={onBack}>
-                    Back
-                </BackButton>
+            <FieldLabel item={field} />
+
+            {localStorage.getItem('dashboard_filters') ? (
```

Comment:
will we remove this before release?

Reply:
I don't think so, because we still need to fix few issues with UI + there are some follow up tasks.